Language/Java/Syntax/IdentCollection.hs

```diff
@@ -0,0 +1,52 @@
+module Language.Java.Syntax.IdentCollection
+  ( IdentCollection (..),
+    addLocalVars,
+    addToClassTrees,
+    addToFields,
+    addToFormalParams,
+    addToTypeNames,
+    isExpressionIdent,
+    isTypeIdent,
+    empty,
+  )
+where
+
+import Language.Java.Syntax.ClassInfo (ClassInfo)
+import Language.Java.Syntax.Equality
+  ( EqOptions (IgnoreSourceSpan),
+    Equality (eq),
+  )
+import Language.Java.Syntax.Types (Ident)
+
+-- | data type used to collect the identifiers that are currently in scope
+data IdentCollection = IdentCollection
+  { icFields :: [Ident],
+    icFormalParams :: [Ident],
+    icLocalVars :: [Ident],
+    icTypeNames :: [Ident],
```

Comment:
Wie besprochen statt `TypeNames` nur noch `Imports`, da die lokalen Types schon in `ClassInfo` stehen